components/com_finder/src/Model/SearchModel.php

```diff
@@ -259,6 +259,13 @@ protected function getListQuery()
 
         $query->order('ordering ' . $db->escape($direction));
 
+        /*
+         * Prevent invalid records from being returned in the final query.
+         * This can happen if the search results are queried while the indexer is running.
+         */
+        $query->where($db->quoteName('object') . ' != ' . $db->quote(''))
+            ->where($db->quoteName('object') . ' IS NOT NULL');
```

Comment:
```suggestion
        $query->where($db->quoteName('object') . ' != ' . $db->quote(''));
```
I've meanwhile checked on MySQL 8 and PostgreSQL and it seems the check for NULL is really not needed there either. So it can be reverted.

Reply:
Sounds good, I have gone ahead and reverted the change, but kept the usage of quoteName.